src/foam/lib/json/ClassReferenceParser.java

```diff
@@ -42,38 +41,16 @@ public PStream parse(PStream ps, ParserContext x) {
   }
 
   public PStream parse(PStream ps, ParserContext x) {
-    ps = super.parse(ps, x);
-
-    if ( ps != null ) {
-      String classId  = (String) x.get("forClass_");
-      String propName = (String) x.get("name");
-
-      Class cls;
-      try {
-        cls = Class.forName(classId);
-      } catch(ClassNotFoundException e) {
-        throw new RuntimeException(e);
-      }
-
-      // TODO(adamvy): Use the context to resolve the class rather than reflection
-      // TODO(adamvy): Better handle errors.
-
-      ClassInfo info;
-      try {
-        info = (ClassInfo) cls.getMethod("getOwnClassInfo").invoke(null);
-      } catch(Exception e) {
-        throw new RuntimeException(e);
+    try {
+      if ( ( ps = super.parse(ps, x)) == null ) {
+        return null;
       }
 
-      Object axiom = info.getAxiomByName(propName);
-
-      if ( axiom == null ) {
-        System.err.println("Unknown Property Reference: " + classId + "." + propName);
-      }
-
-      return ps.setValue(axiom);
+      String classId = (String) x.get("forClass_");
+      Class cls = Class.forName(classId);
+      return cls != null ? ps.setValue(cls) : null;
```

Comment:
Should we be returning a ClassInfo here or just a class?

Reply:
ClassInfo I think. But it happens to work for our current use because INSTANCE_OF will take both.

Reply:
Which can be looked up like this:
    try {
      return INSTANCE_OF((ClassInfo) cls.getMethod("getOwnClassInfo").invoke(null));
    } catch(NoSuchMethodException|IllegalAccessException|InvocationTargetException e) {
      throw new RuntimeException("Attempt to call INSTANCE_OF on non Modelled class." + cls);
    }

Reply:
Yeah I imagine things might act up if we're trying to set a Class property to an actual java class rather than a ClassInfo because the javaType on the Class property is ClassInfo.